server/src/main/java/org/opensearch/action/search/TransportCreatePITAction.java

```diff
@@ -0,0 +1,341 @@
+        SearchRequest searchRequest = new SearchRequest(request.getIndices());
+        searchRequest.preference(request.getPreference());
+        searchRequest.routing(request.getRouting());
+        searchRequest.indicesOptions(request.getIndicesOptions());
+        searchRequest.allowPartialSearchResults(request.isAllowPartialPitCreation());
+
+        final StepListener<SearchResponse> createPitListener = new StepListener<>();
+
+        final ActionListener<SearchResponse> updatePitIdListener = new ActionListener<>() {
+            @Override
+            public void onResponse(SearchResponse searchResponse) {
+                listener.onResponse(searchResponse);
+            }
+
+            @Override
+            public void onFailure(Exception e) {
+                listener.onFailure(e);
+            }
+        };
+
+        /**
+         * Phase 1 of create PIT request : Create PIT reader contexts in the associated shards with a
+         * temporary keep alive
+         */
+        transportSearchAction.executeRequest(task, searchRequest, CREATE_PIT, true, new TransportSearchAction.SinglePhaseSearchAction() {
+            @Override
+            public void executeOnShardTarget(
+                SearchTask searchTask,
+                SearchShardTarget target,
+                Transport.Connection connection,
+                ActionListener<SearchPhaseResult> searchPhaseResultActionListener
+            ) {
+                transportService.sendChildRequest(
+                    connection,
+                    SearchTransportService.CREATE_READER_CONTEXT_ACTION_NAME,
+                    new CreateReaderContextRequest(target.getShardId(), CREATE_PIT_TEMPORARY_KEEP_ALIVE),
+                    searchTask,
+                    new TransportResponseHandler<CreateReaderContextResponse>() {
+                        @Override
+                        public CreateReaderContextResponse read(StreamInput in) throws IOException {
+                            return new CreateReaderContextResponse(in);
+                        }
+
+                        @Override
+                        public void handleResponse(CreateReaderContextResponse response) {
+                            searchPhaseResultActionListener.onResponse(response);
+                        }
+
+                        @Override
+                        public void handleException(TransportException exp) {
+                            searchPhaseResultActionListener.onFailure(exp);
+                        }
+
+                        @Override
+                        public String executor() {
+                            return ThreadPool.Names.GENERIC;
+                        }
+                    }
+                );
+            }
+        }, createPitListener);
+
+        /**
+         * Phase 2 of create PIT : Update PIT reader context with PIT ID and keep alive from request
```

Comment:
Let's mention that PIT will be deleted explicitly if not all PIT contexts are updated successfully in phases 2.